File: driver_flutter/customer/backend/src/modules/pricing/domain/ride-category.ts
export enum RideCategoryType {
  BIKE = 'BIKE',
  BIKE_LITE = 'BIKE_LITE',
  AUTO = 'AUTO',
  AUTO_LITE = 'AUTO_LITE',
  CAB = 'CAB',
  CAB_LITE = 'CAB_LITE',
  PRIME_SEDAN = 'PRIME_SEDAN',
  SEVEN_SEATER = 'SEVEN_SEATER',
}

export interface RideCategory {
  id: string;
  code: RideCategoryType;
  displayName: string;
  description: string;
  capacity: number;
  etaMinutes: number;
  luggageCapacity?: number;
}

export const CATEGORIES: Record<RideCategoryType, RideCategory> = {
  [RideCategoryType.BIKE]: {
    id: 'cat_bike',
    code: RideCategoryType.BIKE,
    displayName: 'Bike',
    description: 'Quickest single-rider bike trip',
    capacity: 1,
    etaMinutes: 2,
    luggageCapacity: 1,
  },
  [RideCategoryType.BIKE_LITE]: {
    id: 'cat_bike_lite',
    code: RideCategoryType.BIKE_LITE,
    displayName: 'Bike lite',
    description: 'Budget-friendly quick bike ride',
    capacity: 1,
    etaMinutes: 3,
    luggageCapacity: 1,
  },
  [RideCategoryType.AUTO]: {
    id: 'cat_auto',
    code: RideCategoryType.AUTO,
    displayName: 'Auto',
    description: 'Doorstep 3-seater auto rickshaw',
    capacity: 3,
    etaMinutes: 4,
    luggageCapacity: 2,
  },
  [RideCategoryType.AUTO_LITE]: {
    id: 'cat_auto_lite',
    code: RideCategoryType.AUTO_LITE,
    displayName: 'Auto lite',
    description: 'Economical pocket-friendly auto',
    capacity: 3,
    etaMinutes: 5,
    luggageCapacity: 2,
  },
  [RideCategoryType.CAB]: {
    id: 'cat_cab',
    code: RideCategoryType.CAB,
    displayName: 'Cab',
    description: 'Comfortable hatchback AC cab',
    capacity: 4,
    etaMinutes: 3,
    luggageCapacity: 2,
  },
  [RideCategoryType.CAB_LITE]: {
    id: 'cat_cab_lite',
    code: RideCategoryType.CAB_LITE,
    displayName: 'Cab lite',
    description: 'Low fare everyday hatchback ride',
    capacity: 4,
    etaMinutes: 4,
    luggageCapacity: 2,
  },
  [RideCategoryType.PRIME_SEDAN]: {
    id: 'cat_prime_sedan',
    code: RideCategoryType.PRIME_SEDAN,
    displayName: 'Prime sedan',
    description: 'Top-rated spacious sedan (Dzire, Etios)',
    capacity: 4,
    etaMinutes: 3,
    luggageCapacity: 3,
  },
  [RideCategoryType.SEVEN_SEATER]: {
    id: 'cat_7_seater',
    code: RideCategoryType.SEVEN_SEATER,
    displayName: '7 seter',
    description: 'Spacious 7-seater SUV (Ertiga, Innova)',
    capacity: 7,
    etaMinutes: 5,
    luggageCapacity: 5,
  },
};

File: driver_flutter/customer/frontend/lib/core/pricing/domain/ride_category.dart
enum RideCategoryType {
  bike,
  bikeLite,
  auto,
  autoLite,
  cab,
  cabLite,
  primeSedan,
  sevenSeater,
}

class RideCategory {
  final String id;
  final RideCategoryType code;
  final String displayName;
  final String description;
  final int capacity;
  final int etaMinutes;

  const RideCategory({
    required this.id,
    required this.code,
    required this.displayName,
    required this.description,
    required this.capacity,
    this.etaMinutes = 3,
  });

  factory RideCategory.fromJson(Map<String, dynamic> json) {
    final codeStr = (json['code'] as String? ?? '').toUpperCase();
    RideCategoryType code = RideCategoryType.cab;

    if (codeStr == 'BIKE') {
      code = RideCategoryType.bike;
    } else if (codeStr == 'BIKE_LITE') {
      code = RideCategoryType.bikeLite;
    } else if (codeStr == 'AUTO') {
      code = RideCategoryType.auto;
    } else if (codeStr == 'AUTO_LITE') {
      code = RideCategoryType.autoLite;
    } else if (codeStr == 'CAB' || codeStr == 'MINI') {
      code = RideCategoryType.cab;
    } else if (codeStr == 'CAB_LITE') {
      code = RideCategoryType.cabLite;
    } else if (codeStr == 'PRIME_SEDAN' || codeStr == 'SEDAN') {
      code = RideCategoryType.primeSedan;
    } else if (codeStr == 'SEVEN_SEATER' || codeStr == 'PREMIUM_XL') {
      code = RideCategoryType.sevenSeater;
    }

    return RideCategory(
      id: json['id'] as String? ?? 'cat_1',
      code: code,
      displayName: json['displayName'] as String? ?? 'Cab',
      description: json['description'] as String? ?? 'Comfortable AC Ride',
      capacity: (json['capacity'] as num?)?.toInt() ?? 4,
      etaMinutes: (json['etaMinutes'] as num?)?.toInt() ?? 3,
    );
  }
}
